xbmc/pvr/PVRDatabase.cpp

```diff
@@ -430,17 +430,17 @@ bool CPVRDatabase::Delete(const CPVRProvider& provider)
   return DeleteValues("providers", filter);
 }
 
-bool CPVRDatabase::Get(CPVRProviders& results,
-                       const std::vector<std::shared_ptr<CPVRClient>>& clients) const
+int CPVRDatabase::Get(CPVRProviders& results,
+                      const std::vector<std::shared_ptr<CPVRClient>>& clients)
 {
-  bool bReturn = false;
+  std::string strQuery = PrepareSQL("SELECT max(idProvider) as maxProviderId from providers");
+  std::unique_lock<CCriticalSection> lock(m_critSection);
+  int iMaxId = GetSingleValueInt(strQuery);
```

Comment:
Why can’t you obtain this value when you actually need it in CPVRProviders?

Reply:
Because I actually need it as a follow on from this call.